cmd/kubeadm/app/cmd/reset.go

```diff
@@ -48,17 +50,24 @@ func NewCmdReset(out io.Writer) *cobra.Command {
 	return cmd
 }
 
-func RunReset(out io.Writer, cmd *cobra.Command, skipPreFlight bool) error {
+type Reset struct{}
+
+func NewReset(skipPreFlight bool) (*Reset, error) {
 	if !skipPreFlight {
 		fmt.Println("Running pre-flight checks")
 		err := preflight.RunResetCheck()
 		if err != nil {
-			return &preflight.PreFlightError{Msg: err.Error()}
+			return nil, &preflight.PreFlightError{Msg: err.Error()}
```

Comment:
> Are you planning to add these everywhere, so that we have init.Run() and join.Run(), etc?

Yes, or at least in the cases where there's pre-flight checks. This will make tests easier to implement, as well.